ross/rotor_assembly.py

```diff
@@ -198,7 +198,7 @@ def flatten(l):
                 disk.tag = "Disk " + str(i)
 
         for i, brg in enumerate(bearing_elements):
-            if brg.__class__.__name__ == "BearingElement" and brg.tag is None:
+            if isinstance(brg, BearingElement) and brg.tag is None:
                 brg.tag = "Bearing " + str(i)
             if brg.__class__.__name__ == "SealElement" and brg.tag is None:
```

Comment:
A bearing instance can be a BearingElement and a SealElement at the same time. To avoid this kind of problem, we have to make something like:

```
if not isinstance(brg, SealElement) and brg.tag is None:
    brg.tag = "Bearing " + str(i)
elif isinstance(brg, SealElement) and brg.tag is None:
    brg.tag = "Seal " + str(i)
```

Reply:
Thanks for the tip, we did try some other methods to check the DoF number of each one, but ended up missing this detail.